### PR TITLE
improve instructions about merging in the step-by-step guide

### PR DESCRIPTION
**Pull request by taleinat**
See the [original discussion on discuss.python.org](https://discuss.python.org/t/info-rebase-origin-maseter-push-f-workflow-corrupts-pull-request-rarely/2558/4?u=taleinat).

This adds detailed instructions on how to deal with merges in two common cases:

1. Pulling other people's commits to your PR branch.
2. Merging in the base (master) branch to resolve merge conflicts.

I've tried to keep to the general tone of the guide, and specifically the "step-by-step" section, keeping things simple, short and straightforward.

### CODE REVIEW

File: pullrequest.rst [hunk 1]
```diff
@@ -114,6 +108,21 @@ You should have already :ref:`set up your system <setup>`,
    git commit -m '<message>'
    git push origin <branch-name>
 
+  * If someone else added new changesets and you get an error::
+
+     git checkout <branch-name>
+     git pull origin <branch-name>  # pull = fetch + merge
+     # if there are merge conflicts: see "Resolving Merge Conflicts" below
+     git push origin <branch-name>
```
Comment (methane) [inline pullrequest.rst]: I think this part is unclear.

* Where someone else added new changesets to?
* What type of error you got?

Comment (terryjreedy) [reply pullrequest.rst]: Moving this block from where is was is an improvement, but I think the following is clearer and fits into the new context even better.
```
* If a core developer reviewing your PR pushed a commit to your PR branch, then add

  git pull origin <branch-name>  # pull = fetch + merge

  after checking out your branch and before editing.  If you have made local changes that have not been pushed
  to your fork and there are merge conflicts, see "Resolving Merge Conflicts" below.
```

Comment (taleinat) [reply pullrequest.rst]: Thanks, those are good comments! I've pushed a change, let me know if you think it's good.

Comment (terryjreedy) [reply pullrequest.rst]: Good enough!

**Pull request merged**